src/features/projectsV2/ProjectListControls/microComponents/ViewModeTabs.tsx

```diff
@@ -1,49 +1,58 @@
-import themeProperties from '../../theme/themeProperties';
-import { useState, ReactNode } from 'react';
-import style from './ProjectListControls.module.scss';
-import ListIcon from '../icons/ListIcon';
-import { SetState } from '../../features/common/types/common';
-import LocationIconOutline from '../icons/LocationIconOutline';
+import { ReactNode } from 'react';
 import { useTranslations } from 'next-intl';
+import themeProperties from '../../../../theme/themeProperties';
+import styles from '../styles/ProjectListControls.module.scss';
+import ListIcon from '../../../../../public/assets/images/icons/projectV2/ListIcon';
+import { SetState } from '../../../common/types/common';
+import LocationIconOutline from '../../../../../public/assets/images/icons/projectV2/LocationIconOutline';
+import { ViewMode } from '../../../../../pages/_app';
 
 interface ViewModeTabsProps {
-  setIsFilterOpen: SetState<boolean>;
-  isSearching: boolean;
+  setIsFilterOpen: SetState<boolean> | undefined;
+  isSearching: boolean | undefined;
+  setSelectedMode: SetState<ViewMode>;
+  selectedMode: ViewMode;
 }
 
-type Tabs = 'list' | 'map';
 interface TabItemProps {
-  selectedTab: Tabs;
+  selectedTab: ViewMode;
   icon: ReactNode;
   label: string | undefined;
 }
-const ViewModeTabs = ({ setIsFilterOpen, isSearching }: ViewModeTabsProps) => {
+const ViewModeTabs = ({
+  setIsFilterOpen,
+  isSearching,
+  setSelectedMode,
+  selectedMode,
+}: ViewModeTabsProps) => {
   const { dark, light } = themeProperties;
   const t = useTranslations('AllProjects');
-  const [selectedMode, setSelectedMode] = useState<'list' | 'map'>('list');
-  const selectTab = (tab: Tabs) => {
-    setIsFilterOpen(false);
+
+  const selectTab = (tab: ViewMode) => {
+    if (setIsFilterOpen) {
+      setIsFilterOpen(false);
+    }
     setSelectedMode(tab);
   };
 
   const TabItem = ({ selectedTab, icon, label }: TabItemProps) => {
     const isSelected = selectedMode === selectedTab;
     const tabButtonClass = isSelected
-      ? style.selectedTabButton
-      : style.unselectedTabButton;
+      ? styles.selectedTabButton
+      : styles.unselectedTabButton;
     return (
       <button className={tabButtonClass} onClick={() => selectTab(selectedTab)}>
         {icon}
-        <div className={style.label}>{label}</div>
+        <div className={styles.label}>{label}</div>
       </button>
     );
   };
 
   const tabContainerClass = isSearching
-    ? style.tabContainerSecondary
-    : style.tabContainer;
+    ? styles.tabContainerSecondary
+    : styles.tabContainer;
 
-  const getIconColor = (mode: Tabs, selectMode: Tabs) =>
+  const getIconColor = (mode: ViewMode, selectMode: ViewMode) =>
```

Comment:
The getIconColor function could be moved outside the component to avoid recreating it on each render.